### PR TITLE
Matrix mult

### LINKED ISSUE #8: Implement matrix multiplication
Implement matrix multiplication using operator overloading:
- New method (operator*) should return a new matrix as a result of multiplication: M = M1 * M2
- Second method (operator*=) should modify current matrix as a result of multiplication: M1 *= M2
- These methods should throw an exception if dimensions of M1 and M2 are not correct.
- Remember: M1 * M2 != M2 * M1
- Test your code by adding new test cases.

### PR DESCRIPTION
**Pull request by varuntumbe**
I have implemented both  the tasks.

### CODE REVIEW

File: MatrixAlgorithms/Matrix2d.cpp [hunk 1]
```diff
@@ -73,6 +73,10 @@ Matrix2d::Matrix2d(std::size_t rows, std::size_t cols, FillType fillType)
     }
 }
 
+
+
+
+
```
Comment (TomaszWychocki) [inline MatrixAlgorithms/Matrix2d.cpp]: Remove empty lines

File: MatrixAlgorithms/Matrix2d.cpp [hunk 2]
```diff
@@ -175,3 +254,6 @@ std::ostream& operator<<(std::ostream& os, const Matrix2d& matrix)
 
     return os;
 }
+
+
+
```
Comment (TomaszWychocki) [inline MatrixAlgorithms/Matrix2d.cpp]: Remove empty lines

File: MatrixAlgorithms/Matrix2d.h [hunk 3]
```diff
@@ -46,11 +49,11 @@ class Matrix2d
...
+    std::size_t rows;
+    std::size_t cols;
     std::vector<std::vector<double>> numbersArray;
 
     /*
      * METHODS
      */
-};
\ No newline at end of file
+};
```
Comment (TomaszWychocki) [inline MatrixAlgorithms/Matrix2d.h]: Not needed

File: MatrixAlgorithms/MatrixAlgorithms.cpp [hunk 4]
```diff
@@ -153,4 +215,8 @@ int main()
     TEST_006_matrix_ones();
     TEST_007_matrix_zeros();
     TEST_008_matrix_identity();
-}
\ No newline at end of file
+    TEST_009_matrix_mult();
+    TEST_010_matrix_mult_assign();
+    TEST_011_matrix_mult();
+    TEST_012_matrix_mult_assign();
```
Comment (TomaszWychocki) [inline MatrixAlgorithms/MatrixAlgorithms.cpp:261]: Please add one more test with 2 matrices filled with random numbers, e.g.:
Matrix2d mat1 = { {1,2,3},{4,5,6} };
Matrix2d mat2 = { {10,11},{20,21},{30,31} };
Matrix2d expectedResult={ {140,146},{320,335} };

File: MatrixAlgorithms/Matrix2d.cpp [hunk 5]
```diff
@@ -135,6 +139,81 @@ void Matrix2d::operator*=(double number)
     }
 }
 
+
+//operator overloading for matrix multiplication (*)
+Matrix2d  Matrix2d::operator * (const Matrix2d& mat2)
```
Comment (TomaszWychocki) [inline MatrixAlgorithms/Matrix2d.cpp:140]: Check and correct the indentation in all files. In this project tab size = 4 spaces.

File: MatrixAlgorithms/Matrix2d.cpp [hunk 6]
```diff
@@ -135,6 +139,81 @@ void Matrix2d::operator*=(double number)
...
+		{
+			//throw an error
+			throw Errors::DIMENSIONAL_ERROR;
+		}
+
+
+		Matrix2d result(this->rows,mat2.cols,FillType::ZEROS);
+
+		//now implement matrix multiplaication algorithm requires o(n^3) complexity
+		for(std::size_t i=0;i<this->rows;i++)
```
Comment (TomaszWychocki) [inline MatrixAlgorithms/Matrix2d.cpp]: Add few spaces:
for(std::size_t i=0; i<this->rows; i++)

**Comment by varuntumbe** [comment]
Okay I will add all these changes.
Thanks.

**Comment by varuntumbe** [comment]
do i have to resolve this conflicts ?

**Comment by TomaszWychocki** [comment]
Yes, of course

**Pull request merged**